L1Trigger/Phase2L1ParticleFlow/interface/taus/L1HPSPFTauEmulator.h

```diff
@@ -0,0 +1,439 @@
+#ifndef L1HPSPFTAUEMULATOR_H
+#define L1HPSPFTAUEMULATOR_H
+
+#include "ap_int.h"
+#include "ap_fixed.h"
+#include <iostream>
+#include <vector>
+#include <numeric>
+#include <algorithm>
+
+namespace L1HPSPFTauEmu {
+
+  //mapping the eta phi onto bits
+
+  static float etaphi_base = 720. / 3.14159;
+  static float dz_base = 0.05;
+  typedef ap_ufixed<16, 14> pt_t;
+
+  typedef ap_int<12> etaphi_t;
+
+  typedef ap_int<12> detaphi_t;
+
+  typedef ap_uint<20> detaphi2_t;
+  typedef ap_uint<5> count_t;  //type for multiplicity
+  typedef ap_uint<3> type_t;   //type for particle type
+  typedef ap_int<10> dz_t;     //type for z0 cut
+
+  class Particle {
+  public:
+    pt_t hwPt;
+    etaphi_t hwEta;
+    etaphi_t hwPhi;
+    type_t hwId;
+    dz_t hwZ0;
+
+    Particle() : hwPt(0), hwEta(0), hwPhi(0), hwId(0), hwZ0() {}
+    Particle(pt_t pt, etaphi_t eta, etaphi_t phi, type_t id, dz_t z0)
+        : hwPt(pt), hwEta(eta), hwPhi(phi), hwId(id), hwZ0(z0) {}
+
+    pt_t pt() const { return hwPt; }
+
+    etaphi_t phi() const { return hwPhi; }
+
+    etaphi_t eta() const { return hwEta; }
+    type_t id() const { return hwId; }
+    dz_t z0() const { return hwZ0; }
+  };
+
+  //template classes
+  class Tau : public Particle {
+  public:
+    Tau(pt_t pt, etaphi_t eta, etaphi_t phi, type_t id, dz_t z0) : Particle(pt, eta, phi, id, z0) {}
+    ap_uint<5> seed_index;
+    pt_t seed_pt;
+    etaphi_t seed_eta;
+    etaphi_t seed_phi;
+    dz_t seed_z0;
+  };
+
+  //constants
+  static const detaphi_t TWOPI = 3.14159 * 2. * etaphi_base;
+  static const detaphi_t PI = 3.14159 * etaphi_base;
```

Comment:
Silly question, but is there some reason we use pi to this precision instead of some standard library representation?